### PR TITLE
VIAMW-604 Automatize Appgrid entry type updates

### PR DESCRIPTION
**Pull request by AlinStoianAccedo**

### CODE REVIEW

File: src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/ChangeSet_1_3.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class ChangeSet_1_3 extends AbstractChangeSet {
+
+    @Override
+    public void run(){
+        String applicationGroupId = ApplicationGroupType.VIA_GO_STAGING.getApplicationGroupId();
```
Comment (Erikbohlmark) [inline src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/ChangeSet_1_3.java]: This should be received as an argument. Cant have it as an enum then cause we don't know the application group id. should be able to take an array of ids :)

File: src/main/java/tv/accedo/via/middleware/appgrid/tools/Main.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+@Slf4j
+public class Main {
+
+    private static final Map<String, Class<? extends Runnable>> taskMapping = new HashMap<>();
+
+    static {
+        // Add here additional change sets
+        taskMapping.put("test", TestConnection.class);
+        taskMapping.put("1.3", ChangeSet_1_3.class);
```
Comment (Erikbohlmark) [inline src/main/java/tv/accedo/via/middleware/appgrid/tools/Main.java]: They should be in a ordered list. and you are able to define which version you want to upgrade from and to.

File: src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/AbstractChangeSet.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+@Slf4j
+public abstract class AbstractChangeSet implements Runnable {
+
+    private final AppgridClient appgridClient = new AppgridClient();
+    final ConvertorService convertorService = new ConvertorService();
+
+    AbstractChangeSet() {
+        try {
+            LOG.info("Logging in to Appgrid");
+            appgridClient.login();
```
Comment (Erikbohlmark) [inline src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/AbstractChangeSet.java]: Do we login for every changeset?

File: src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/ChangeSet_1_3.java (hunk 1 above)
Comment (AlinStoianAccedo) [inline src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/ChangeSet_1_3.java]: Yes, it should be an argument...

File: src/main/java/tv/accedo/via/middleware/appgrid/tools/Main.java (hunk 2 above)
Comment (AlinStoianAccedo) [inline src/main/java/tv/accedo/via/middleware/appgrid/tools/Main.java]: The tool was meant to be run once for each change set. So the order is defined outside of the tool.

File: src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/AbstractChangeSet.java (hunk 3 above)
Comment (AlinStoianAccedo) [inline src/main/java/tv/accedo/via/middleware/appgrid/tools/changeset/AbstractChangeSet.java]: Yes, as for now it only supports a single change set per execution

**Pull request merged**